app/views/landing-pages/parents/ModalTimetapSchedule.vue

```diff
@@ -0,0 +1,105 @@
+<script>
+  import BaseModal from 'app/components/common/BaseModal'
+
+  export default {
+    components: {
+      BaseModal
+    },
+
+    props: {
+      show: {
+        type: Boolean,
+        default: false
+      },
+
+      classType: {
+        type: String,
+        default: undefined
+      }
+    },
+
+    computed: {
+      timetapIframeUrl () {
+        if (this.classType === undefined) {
+          return 'https://codecombat.timetap.com?utm_campaign=timetapliveclasses&utm_source=codecombat&utm_medium=modal'
+        } else if (this.classType === 'group') {
+          return 'https://www.timetap.com/appts/jPlOQTv7JXIJ?utm_campaign=timetapliveclasses&utm_source=codecombat&utm_medium=modal'
+        } else if (this.classType === 'private') {
+          return 'https://www.timetap.com/appts/wPvlbTkKwauE?utm_campaign=timetapliveclasses&utm_source=codecombat&utm_medium=modal'
+        }
+      }
+    },
+
+    methods: {
+      onWindowMessage (event) {
+        if (event.source === this.$refs.timetapIframe.contentWindow && event.data === 'class.booked') {
+          this.$emit('booked')
+        }
+      }
+    },
+
+    mounted () {
+      window.addEventListener('message', this.onWindowMessage, false)
+    },
+
+    beforeDestroy () {
+      window.removeEventListener('message', this.onWindowMessage)
+    }
+  }
+</script>
+
+<template>
+  <base-modal :class="[ !show ? 'hide' : undefined, 'timetap-modal']">
+    <template slot="header">
+      {{ $t('parents_landing_2.book_your_class') }}
+      <span class="glyphicon glyphicon-remove close" @click="$emit('close')"></span>
+    </template>
+
+    <template slot="body">
+      <iframe ref="timetapIframe" :src="timetapIframeUrl" />
+    </template>
+  </base-modal>
+</template>
+
+<style scoped>
+  .hide {
+    display: none;
+  }
+
+  .timetap-modal ::v-deep .modal-container {
```

Comment:
I really like that the `v-deep` combinator lets you add scoped css with nesting.